index.js

```diff
@@ -59,6 +86,7 @@ module.exports = function(fn) {
 
     res.end = function(chunk, encoding, cb) {
       debug('end called');
+      endCalled = true;
```

Comment:
It is not possible to support the reason argument in 0.10 at least. In node 4.0 it can be done though. This is how we did it in express-hijackresponse, and that haven't been problematic for us until now.